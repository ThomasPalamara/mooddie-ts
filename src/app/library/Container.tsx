import React from 'react';
import styled from '@emotion/styled';

interface Props {
  className?: string;
}

const Container: React.FunctionComponent<Props> = ({ children, ...others }) => {
  const Element = styled.div`
    container {
      width: 100%;
      margin-left: auto;
      margin-right: auto;
    }
    @media (min-width: 640px) {
      .container {
        max-width: 640px;
      }
    }
    @media (min-width: 768px) {
      .container {
        max-width: 768px;
      }
    }
    @media (min-width: 1024px) {
      .container {
        max-width: 1024px;
      }
    }
    @media (min-width: 1280px) {
      .container {
        max-width: 1280px;
      }
    }
  `;

  return <Element {...others}>{children}</Element>;
};

export default Container;
