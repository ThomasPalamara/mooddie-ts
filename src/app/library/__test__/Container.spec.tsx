import React from 'react';
import { shallow } from 'enzyme';
import { merge } from 'lodash';

import Container from '../Container';

const setup = (overrides = {}) => {
  const props = merge({}, overrides);
  const wrapper = shallow(<Container {...props} />);

  return { wrapper, props };
};
describe('Container', () => {
  it('should match Snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
