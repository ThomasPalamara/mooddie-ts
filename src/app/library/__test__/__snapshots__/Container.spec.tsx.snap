// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match Snapshot 1`] = `<Styled(div) />`;
